prov/gni/src/gnix_fabric.c

```diff
@@ -636,8 +634,10 @@ static int _gnix_ep_getinfo(enum fi_ep_type ep_type, uint32_t version,
 					goto err;
 				}
 			} else {
+				/* FIXME: Check will not work with ofi_check_mr_mode
 				if (__gnix_check_mr_mode(mr_mode))
 					goto err;
+				*/
```

Comment:
I cannot test the gni provider with these changes.  Note that the __gnix_check_mr_mode test has been commented out.  Something needs to be done here.  In theory the call to ofi_check_mr_mode() should be sufficient to validate whether the gni provider can support the request.  Please see if that call is sufficient and the gni specific call can be removed.